### PR TITLE
Adoption to new easyscience (fitting, parameter unique name 

### PR DESCRIPTION
**Pull request by andped10**
No intended changes to functionality only adoption to the new easyscience lib.

Main changes are
- `raw_value` -> `value`
- `uid` -> `unique_name`
- Various updates to `as_dict` and `from_dict`
- introduced new variable `populate_if_none` to prevent the creation of internal variables when creating an object using `from_dict`
- `unique_names` are determined for internal variables.

When testing:
- Need to clean global object when doing `dict_round_trip` (`as_dict` and `from_dict`) tests
- Need to sort the produced data dict when doing `as_dict` and `from_dict` tests
- Changes to the produced unit 
- Not sure if parameterized tests are stable when the global object is cleaned

### CODE REVIEW

File: src/easyreflectometry/sample/assemblies/gradient_layer.py [hunk 1]
```diff
@@ -100,8 +101,8 @@ def roughness(self, roughness: float) -> None:
     def _dict_repr(self) -> dict[str, str]:
         """A simplified dict representation."""
         return {
-            'thickness': self.thickness,
-            'discretisation_elements': self._discretisation_elements,
+            'thickness': float(self.thickness),
+            'discretisation_elements': int(self._discretisation_elements),
```
Comment (rozyczko) [inline src/easyreflectometry/sample/assemblies/gradient_layer.py:105]: why are those explicit coversions needed?

**Review by rozyczko — APPROVED** [review]
This seems to look OK, I had several general questions, but nothing prohibiting from the merge.

File: src/easyreflectometry/experiment/model_collection.py [hunk 2]
```diff
@@ -19,13 +20,20 @@ def __init__(
...
-            models = [Model(interface=interface) for _ in range(SIZE_DEFAULT_COLLECTION)]
+            if populate_if_none:
+                models = [Model(interface=interface) for _ in range(SIZE_DEFAULT_COLLECTION)]
+            else:
+                models = []
         super().__init__(name, interface, *models, **kwargs)
         self.interface = interface
 
+        # Needed by the as_dict functionality
+        self.populate_if_none = False
```
[the remark this answers is not in the document]
Comment (andped10) [reply src/easyreflectometry/experiment/model_collection.py]: added code comment:
        # Needed to ensure an empty list is created when saving and instatiating the object as_dict -> from_dict
        # Else collisions might occur in global_object.map

File: src/easyreflectometry/sample/assemblies/gradient_layer.py (hunk 1 above)
Comment (andped10) [reply src/easyreflectometry/sample/assemblies/gradient_layer.py]: PR response
These are properties that else appears weird in the dictionary

**Pull request merged**